### PR TITLE
linphone: enable gtk ui

### PR DESCRIPTION
**Pull request by stesie**
###### Motivation for this change

Linphone voip client currently depends on Gtk, yet the Gtk-based UI is *not* currently available.  This passes `ENABLE_GTK_UI=ON` to actually enable it.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/linphone/default.nix [hunk 1]
```diff
@@ -18,6 +18,12 @@ stdenv.mkDerivation rec {
     sha256 = "0az2ywrpx11sqfb4s4r2v726avcjf4k15bvrqj7xvhz7hdndmh0j";
   };
 
+  cmakeFlags = "-DENABLE_GTK_UI=ON";
+
+  postPatch = ''
+    touch coreapi/liblinphone_gitversion.h
```
Comment (timokau) [inline pkgs/applications/networking/instant-messengers/linphone/default.nix]: Why is this necessary? What does it do?

File: pkgs/applications/networking/instant-messengers/linphone/default.nix [hunk 2]
```diff
@@ -18,6 +18,12 @@ stdenv.mkDerivation rec {
     sha256 = "0az2ywrpx11sqfb4s4r2v726avcjf4k15bvrqj7xvhz7hdndmh0j";
   };
 
+  cmakeFlags = "-DENABLE_GTK_UI=ON";
```
Comment (timokau) [inline pkgs/applications/networking/instant-messengers/linphone/default.nix]: Its better do use lists for `*Flags` (except some very specific cases)

**Review by timokau — DISMISSED** [review]
Does this increase the closure size? If it has any negative consequences it should probably be made an option instead, though it may be enabled by default.

File: pkgs/applications/networking/instant-messengers/linphone/default.nix (hunk 1 above)
Comment (stesie) [reply pkgs/applications/networking/instant-messengers/linphone/default.nix]: The source file `gtk/main.c` depends on this file.  Linphone actually has two build systems, one based on cmake, the other one based on autoconf/automake.  The `coreapi/Makefile.am` file actually provides the `liblinphone_gitversion.h` file (by just touching it), however cmake doesn't.  Since otherwise the build would now fail I just touch the file (like `coreapi/Makefile.am` would)

**Comment by stesie** [comment]
This change indeed increases the closure size (by roughly 3MB):

```
$ nix path-info -S /nix/store/3cdpaigzk2dlngmd2mjk13fl0nds4ij2-linphone-3.12.0
/nix/store/3cdpaigzk2dlngmd2mjk13fl0nds4ij2-linphone-3.12.0	  829181032
$ nix path-info -S /nix/store/b92zfm3qqxjhqcqm298j92zjg7qq7jyp-linphone-3.12.0
/nix/store/b92zfm3qqxjhqcqm298j92zjg7qq7jyp-linphone-3.12.0	  832354528
```

... I've just added a `withGui` option defaulting to `true`, that allows to disable it

File: pkgs/applications/networking/instant-messengers/linphone/default.nix (hunk 2 above)
Comment (stesie) [reply pkgs/applications/networking/instant-messengers/linphone/default.nix]: done.

File: pkgs/applications/networking/instant-messengers/linphone/default.nix (hunk 1 above)
Comment (timokau) [reply pkgs/applications/networking/instant-messengers/linphone/default.nix]: That sounds like a bug, can you please open an upstream ticket for that and link it in a comment?

**Comment by timokau** [comment]
Thanks! While 3MB really aren't much here, I think making it clear that it is optional is still better style. With it being enabled by default, we have the best of both worlds.

**Comment by timokau** [comment]
It wasn't entirely obsolete:

> That sounds like a bug, can you please open an upstream ticket for that and link it in a comment?

We shouldn't just quietly patch around bugs, they at least need to be reported upstream.

**Comment by stesie** [comment]
@timokau well this has been "fixed" in upstream already, ... the file failing to compile doesn't exist any longer. Actually upstream is porting the user interface to Qt currently.
